scripts/generate_package_diffs_for_flavors.py

```diff
@@ -0,0 +1,78 @@
+import sys
+import pandas as pd
+import subprocess
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from io import StringIO
+
+
+def parse_package_list_file(file_path:Path)->str:
+    parse_package_list_file_command = ["perl","ext/scripts/list_newest_versions/extract_columns_from_package_lisl.pl","--file",str(file_path),"--columns","0,1"]
+    result = subprocess.run(parse_package_list_file_command,stdout=subprocess.PIPE)
+    result.check_returncode()
+    result_string = result.stdout.decode("UTF-8")
+    return result_string
+
+def compare_package_lists(package_list_1:str, package_list_2:str)->pd.DataFrame:
+    package_list_1_df = pd.read_csv(StringIO(package_list_1), delimiter="|", names=["package","version"])
+    package_list_2_df = pd.read_csv(StringIO(package_list_2), delimiter="|", names=["package","version"])
+    diff_df = pd.merge(package_list_1_df, package_list_2_df, how='outer', on='package',sort=False)
+    diff_df = diff_df.sort_values("package")
+    new = diff_df["version_y"].isnull()
+    removed = diff_df["version_x"].isnull()
+    updated =  ~diff_df["version_x"].isnull() & (diff_df["version_x"]!=diff_df["version_y"])
+    diff_df["status"]=""
+#    diff_df["status"].loc[new]="NEW"
+#    diff_df["status",removed]="REMOVED"
+    diff_df["status"].values[updated]="UPDATED"
+    diff_df = diff_df.reset_index(drop=True)
+    return diff_df
+
+def compare_build_step(flavor_path:Path, build_step_path:Path, working_copy_1:Path, working_copy_2:Path):
+    packages_path = Path(build_step_path,"packages")
+    if packages_path.is_dir():
+        for package_list_file  in packages_path.iterdir():
+            package_list_working_copy_str_1 = parse_package_list_file(Path(working_copy_1,package_list_file))
+            package_list_working_copy_str_2 = parse_package_list_file(Path(working_copy_2,package_list_file))
+            diff = compare_package_lists(package_list_working_copy_str_1, package_list_working_copy_str_2)
+            print(diff)
+
+
+def compare_flavor(flavor_path:Path, working_copy_1:Path, working_copy_2:Path):
+    flavor_base_path = Path(flavor_path,"flavor_base")
+    if flavor_base_path.is_dir():
+        for build_step_path in flavor_base_path.iterdir():
+            if build_step_path.is_dir():
```

Comment:
Currently you iterate over all paths/files. But the perl script works only with apt files, bot pip package files or cran files, correct?